test_runner/src/main/kotlin/ftl/reports/outcome/CreateMatrixOutcomeSummary.kt

```diff
@@ -0,0 +1,16 @@
+package ftl.reports.outcome
+
+import com.google.api.services.toolresults.model.Environment
+
+fun TestOutcomeContext.createMatrixOutcomeSummary(): Pair<BillableMinutes, TestOutcome> =
+    steps.calculateAndroidBillableMinutes(projectId, testTimeout) to
+        if (environments.isNotEmpty() && environments.hasNoOutcome().not())
+            environments.createMatrixOutcomeSummaryUsingEnvironments(matrixId)
+        else {
+            if (steps.isEmpty()) println("No test results found, something went wrong. Try re-running the tests.")
```

Comment:
I think we could return here just `TestOutcome("Unknown",  matrixId, "Unknown outcome")` (if steps is an empty list)

Reply:
On the second thought, I think this would introduce another 'logic path' to maintain and might worsen readability so I am not that much convinced now...

Reply:
Do not look at `if` with `println` and everything will be fine ;). the `createMatrixOutcomeSummaryUsingEnvironments` and `createMatrixOutcomeSummaryUsingSteps` are safe. The `createMatrixOutcomeSummary` is only glue.

Reply:
Yeah, that's why I wrote the second comment. Actually now, when I see those lines I can't tell why it bothered me at first-time ¯\_(ツ)_/¯